src/main/java/com/odysseusinc/arachne/datanode/service/impl/DataSourceServiceImpl.java

```diff
@@ -239,7 +239,7 @@ public DataSource update(User user, DataSource dataSource) {
         if (Objects.nonNull(atlasTargetCohortTable)) {
             exists.setCohortTargetTable(atlasTargetCohortTable);
         }
-        if (DBMSType.IMPALA.equals(type)) {
+        if (DBMSType.IMPALA.equals(type) || DBMSType.BIGQUERY.equals(type)) {
```

Comment:
How Kerberos relates to BigQuery? If both Impala and BQ can have `keyfile` then why not to wrap only it into a shared `if`? Or maybe even have a strategy? ;)